testkit/scaladsl/src/test/scala/com/lightbend/lagom/scaladsl/testkit/PersistentEntitySuite.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package com.lightbend.lagom.scaladsl.testkit
+
+import akka.actor.ActorSystem
+import akka.testkit.TestKit
+import com.lightbend.lagom.scaladsl.persistence.PersistentEntity
+import com.lightbend.lagom.scaladsl.playjson.JsonSerializerRegistry
+import org.scalatest.{Assertion, BeforeAndAfterAll, TestSuite}
+
+/**
+ * A mixin to provide common PE test facilities like `withDriver` and
+ * actor system shutdown mechanisms.
+ */
+trait PersistentEntitySuite { this: TestSuite with BeforeAndAfterAll =>
+
+  /**
+   * The PE's command type
+   */
+  type EntityCommand
+
+  /**
+   * The PE's event type
+   */
+  type EntityEvent
+
+  /**
+   * The PE's State type
+   */
+  type EntityState
+
+  /**
+   * The name to be given to the actor system.
+   */
+  def actorSystemName: String
+
+  /**
+   * The JSON serializer registry for the given PE.
+   */
+  def serializerRegistry: JsonSerializerRegistry
+
+  protected val system = ActorSystem(actorSystemName, JsonSerializerRegistry.actorSystemSetupFor(serializerRegistry))
+
+  override protected def afterAll(): Unit = {
+    TestKit.shutdownActorSystem(system)
```

Comment:
I don't think it's responsibility of this trait to shutdown the actor system. 🤔

Reply:
Should it then also be the client's responsibility to inject an actor system?

Reply:
I was thinking about just abstracting away the actor system from the client, but if that's not desireable, I can remove it.